lgc/patch/NggPrimShader.h

```diff
@@ -145,9 +204,15 @@ class NggPrimShader {
   llvm::Function *createGsEmitHandler(llvm::Module *module, unsigned streamId);
   llvm::Function *createGsCutHandler(llvm::Module *module, unsigned streamId);
 
-  llvm::Value *readPerThreadDataFromLds(llvm::Type *readDataTy, llvm::Value *threadId, NggLdsRegionType region);
+  llvm::Value *readPerThreadDataFromLds(llvm::Type *readDataTy, llvm::Value *threadId, NggLdsRegionType region,
+                                        bool useDs128 = false);
+  void writePerThreadDataToLds(llvm::Value *writeData, llvm::Value *threadId, NggLdsRegionType region,
+                               bool useDs128 = false);
 
-  void writePerThreadDataToLds(llvm::Value *writeData, llvm::Value *threadId, NggLdsRegionType region);
+  llvm::Value *readVertexCullInfoFromLds(llvm::Type *readDataTy, llvm::Value *vertexItemOffset, unsigned dataOffset,
+                                         bool useDs128 = false);
+  void writeVertexCullInfoToLds(llvm::Value *writeData, llvm::Value *vertexItemOffset, unsigned dataOffset,
+                                bool useDs128 = false);
```

Comment:
These functions are never called with useDs128, and they can't ever reliably called with it because the cull info stride is an odd number of dwords -- right? So better to remove that parameter altogether.

Reply:
I will remove this parameter. It is indeed useless and unsafe.